docs/html/user_guide.rst

```diff
@@ -1306,6 +1306,238 @@ issue tracker`_ if you believe that your problem has exposed a bug in pip.
 
 .. _`Using pip from your program`:
 
+Backtracking user guide information
+===================================
+
+Related to `GH issue 8713 <https://github.com/pypa/pip/issues/8713>`__.
+
+Objective
+---------
+
+The purpose of this content is to provide documentation of 1) why this
+occurs, and 2) how the user can possible solve it.
+
+Location
+--------
+
+The end location of this content will be on\ `the pip user
+guide <https://pip.pypa.io/en/latest/user_guide/#id35>`__, somewhere
+near the Resolution Impossible section.
+
+--------------
+
+Dependency resolution backtracking
+==================================
+
+Or more commonly known as *"Why does pip download multiple versions of
+the same package over and over again during an install?"*.
+
+The purpose of this section is to provide explanation of why
+backtracking happens, and practical suggestions to pip users who
+encounter dependency resolution backtracking during a ``pip install``
+command.
+
+What is backtracking?
+---------------------
+
+Backtracking is not a bug, or an unexpected behaviour. It is part of the
+way pip's dependency resolution process works.
+
+During a pip install (e.g. ``pip install tea``), pip needs to work out
+the packages dependencies (e.g. ``spoon``, ``hot-water``, ``cup`` etc),
+and the versions of each of these packages it needs to install.
+
+For each of these dependent packages, it needs to work out which version
+of a package is a good candidate to install.
```

Comment:
What does "a good candidate" mean? I would say something like "is a good candidate to install. A good candidate, in general, it is compatible with all the other packages the user has already installed, or is installing in the same command, and works with other requirements or constraints the user has specified". @pfmoore is that mostly ok as a one-sentence simplification?

Reply:
Yes, that seems OK to me. Maybe in the longer term, we should have somewhere in the architecture docs that describes *precisely* what a compatible candidate is, and then link to that from here. But as a summary the above is good.

Reply:
In the context of this *backtracking* - a good candidate is specifically versions of these packages that are compatible *1) with the other packages in this install process*? Or *2) with the other packages in this install process and every other package installed in this environment*? From the discussion with @pradyunsg I understood it was the first.

If it is 1) then since this documentation is about backtracking specifically, the information priority should focus on it.

I propose, "A good candidate is compatible with all the other packages versions being installing in the same command, and works with other requirements or constraints the user has specified".

Reply:
Yes, I'm pretty sure it's (1), although there's definitely an argument for (2) - we had that argument in a ticket somewhere, and while I think we decided on (1) it wasn't clear-cut, and I could see the case for changing the behaviour in the future.

Are we over-thinking this? I get the impression that the people concerned with "why is pip downloading loads of versions of this package" probably won't understand (or care about) these nuances...

Reply:
> Are we over-thinking this? I get the impression that the people concerned with "why is pip downloading loads of versions of this package" probably won't understand (or care about) these nuances...

I feel like clarifying what a "good candidate" means is a good idea. I'm not sure what level of detail we should go into here, and I haven't thought about this enough to have an opinion on it.

> we had that argument in a ticket somewhere

#7744 -- why do I remember these numbers. :(